identity/src/secp256k1.rs

```diff
@@ -131,30 +132,30 @@ impl SecretKey {
     ///
     /// [RFC3278]: https://tools.ietf.org/html/rfc3278#section-8.2
     pub fn sign(&self, msg: &[u8]) -> Vec<u8> {
-        let generic_array = Sha256::digest(msg);
+        let digest = Sha256::new_with_prefix(msg);
 
-        // FIXME: Once `generic-array` hits 1.0, we should be able to just use `Into` here.
-        let mut array = [0u8; 32];
-        array.copy_from_slice(generic_array.as_slice());
-
-        let message = Message::parse(&array);
-
-        libsecp256k1::sign(&message, &self.0)
+        self.0
+            .sign_digest_recoverable(digest)
+            .unwrap()
             .0
-            .serialize_der()
-            .as_ref()
-            .into()
+            .to_der()
+            .to_bytes()
+            .to_vec()
```

Comment:
```suggestion
            .to_bytes()
            .into_vec()
```

Converts without allocation.